bundles/org.openhab.binding.modbus.solaxx3mic/README.md

```diff
@@ -0,0 +1,296 @@
+It has been also found that nearly all inverters are publishing the same set of data, but in different order.
+So I have doesn't hardcoded modbus register map inside this binding, but I have put it into thing and channel configuration.
+
+## Supported Things
+
+It has been designed as highly configurable, so at this moment it is confirmed that after additional configuration it is working for Growatt TL3-S Photovoltaic Inverter as well.
+
+Generally binding should be able to handle any one and two channels Inverter what publishes modbus data in a single continuous block of register shorter than 64 registers.
+
+Moreover, this binding should be really easy to extend to other devices with single block of registers, because it is required only to edit thing description xml files from resources\OH-INF folder to transform it to completelly different binding. 
+Java code is written that way, that is scans all defined and mapped to items channels and maps received modbus registers block to them dynamically.
+
+## Discovery
+
+Because of flexibility of this binding discovery is not supported.
+
+## Binding Configuration
+
+Binding does not requires any special configuration.
+
+## Thing Configuration
+
+Thing configuration is necessary in configuration file because of extended amount of information required by thing.
+
+Basically this thing requires Bridge of type Modbus TCP or Modbus Serial.
+It also requires definition of following parameters:
+
+1. refresh - amount of time between polls [s]
+2. inputAddress - begin of modbus type 3 registers block
+3. inputBlockLength - length of modbus type 3 registers block
+4. maxTries - maximal amount of tries during failure condition
+5. holdingAddress - begin of modbus type 4 registers block
+6. holdingBlockLength - length of modbus type 4 registers block
+
+
+## Channels
+
+Channels are splitted into several groups.
+All channels are read only.
+Not all channels are required to use.
+
+Groups:
+
+1. Device information
+2. AC Line General
+3. AC Phases A, B, C as separate Groups
+4. DC Lines 1 and 2 as separate Groups
+
+Channels in every group type:
+
+1. Device Information (device-information)
+   - status (status-type)
+   - cabinet-temperature (cabinet-temperature-type)
+   - heatsink-temperature(heatsink-temperature-type)
+2. AC Summary (ac-general)
+   - ac-power (ac-power-type)
+   - ac-dayly-energy (ac-dayly-energy-type)
+   - ac-lifetime-energy (ac-lifetime-energy-type)
+3. AC Phase (ac-phase)
+   - ac-phase-voltage-to-n (ac-phase-voltage-to-n-type)
+   - ac-phase-current (ac-phase-current-type)
+   - ac-phase-frequency (ac-phase-frequency-type)
+   - ac-phase-power (ac-power-type)
+4. DC summary (dc-general )
+   - dc-current (dc-current-type)
+   - dc-voltage (dc-voltage-type)
+   - dc-power (dc-power-type)
+
+Every channel configuration consists of five parameters:
+
+1. registerFunction - identification of block where register resides.
+   - Possible are:
+     - 3 - Input register
```

Comment:
Detail, consider having this parameter given as string, similar to modbus binding. We would avoid having "cryptic" numbers in the config.